core/src/main/java/jenkins/security/ConfidentialStore.java

```diff
@@ -121,15 +121,21 @@ void clear() {
         }
 
         @Override
-        protected void store(ConfidentialKey key, byte[] payload) throws IOException {
-            LOGGER.fine(() -> "storing " + key.getId() + " " + Util.getDigestOf(Util.toHexString(payload)));
-            data.put(key.getId(), payload);
+        protected void store(ConfidentialKey key, byte[] payload) throws IllegalStateException {
+                byte[] payloadDigest = Util.getSHA256DigestOf(payload);
+                String payloadDigestHexString = (payloadDigest != null) ? Util.toHexString(payloadDigest) : null;
+                //called only from tests
+                LOGGER.fine("storing " + key.getId() + " " + payloadDigestHexString);
+                data.put(key.getId(), payload);
         }
 
         @Override
-        protected byte[] load(ConfidentialKey key) throws IOException {
+        protected byte[] load(ConfidentialKey key) throws IllegalStateException {
```

Comment:
you should not generally (and in this case) declare a `RuntimeException` in a method signature.  
Keep to the parent method signature.